Cineaste/CoreDataHelper/MovieStorage.swift

```diff
@@ -97,12 +97,56 @@ class MovieStorage {
             storedMovie.releaseDate = movie.releaseDate
             storedMovie.runtime = movie.runtime
             storedMovie.voteAverage = movie.voteAverage
+            storedMovie.voteCount = movie.voteCount
 
             storedMovie.watched = watched
+            storedMovie.watchedDate = watched ? Date() : nil
             self.save(handler: handler)
         }
     }
 
+    func insertMoviesFromImport(with movies: [StoredMovie],
+                                handler: ((_ result: Result<Bool>) -> Void)? = nil) {
+
+        let dispatchGroup = DispatchGroup()
+
+        backgroundContext.perform {
```

Comment:
If all StoredMovies are created on the right context they can just be saved without recreating other StoredMovies.
This can lead to duplicate entries or in our case to merge conflicts if the given `StoredMovies` are created also on the backgroundContext. Whenever a `StoredMovie` is handed to a function the function must assume that the movie is already persisted. If a method wants to safe `Movies` it should only accept the `Movie` Entity and not `StoredMovies`. We could just save the `[StoredMovie]`  right within the `importData` method which calls this method.